### PR TITLE
TRAIN: Handle few-shot data and model loading

### PR DESCRIPTION
**Pull request by amitir22**

### CODE REVIEW

**Review by talafek96 — CHANGES_REQUESTED** [review]

File: extern/guided_diffusion/guided_diffusion/image_datasets.py [hunk 1]
```diff
@@ -38,14 +40,27 @@ def load_data(
...
-    if class_cond:
-        # Assume classes are the first part of the filename,
-        # before an underscore.
-        class_names = [bf.basename(path).split("_")[0] for path in all_files]
-        sorted_classes = {x: i for i, x in enumerate(sorted(set(class_names)))}
-        classes = [sorted_classes[x] for x in class_names]
+
+    if is_few_shot and few_shot_count is not None:
+        samples_dir = bf.join(data_dir, _select_random_class())
+        all_files = _random_select_count_samples_from(samples_dir, few_shot_count)
```
[the remark this answers is not in the document]
Comment (amitir22) [reply extern/guided_diffusion/guided_diffusion/image_datasets.py:48]: k

File: extern/guided_diffusion/guided_diffusion/image_datasets.py [hunk 2]
```diff
@@ -67,6 +83,14 @@ def load_data(
         yield from loader
 
 
+def _select_random_class():
+    return 'hazelnut'  # TODO: obviously change to actually select in random
```
[the remark this answers is not in the document]
Comment (amitir22) [reply extern/guided_diffusion/guided_diffusion/image_datasets.py:87]: k

File: extern/guided_diffusion/guided_diffusion/image_datasets.py [hunk 3]
```diff
@@ -67,6 +83,14 @@ def load_data(
...
+
+
+def _random_select_count_samples_from(path: str, count: int):
+    return random.sample(_list_image_files_recursively(path), count)
```
[the remark this answers is not in the document]
Comment (amitir22) [reply extern/guided_diffusion/guided_diffusion/image_datasets.py:91]: k

File: utils/models.py [hunk 4]
```diff
@@ -62,7 +62,7 @@ def __init__(self):
         )
         self.default_args.update(model_and_diffusion_defaults())
 
-    def get_model(self, model_name: str, to_compile: bool=True) -> Tuple[UNetModel, GaussianDiffusion]:
+    def get_model(self, model_name: str, args: dict=None, to_compile: bool=True) -> Tuple[UNetModel, GaussianDiffusion]:
```
[the remark this answers is not in the document]
Comment (amitir22) [reply utils/models.py:65]: ok

File: scripts/train.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    add_dict_to_argparser,
+)
+from guided_diffusion.train_util import TrainLoop
+from utils.models import ModelLoader
+
+
+def main():
+    args = create_argparser().parse_args()
+
+    if type(args.few_shot_count) not in [int, None] or args.few_shot_count is int and args.few_shot_count <= 0:
```
Comment (talafek96) [inline scripts/train.py:23]: put parenthesis around the second condition

```suggestion
    if (type(args.few_shot_count) not in [int, None]) or (args.few_shot_count is int and args.few_shot_count <= 0):
```

**Review by talafek96 — CHANGES_REQUESTED** [review]

File: scripts/train.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+)
+from guided_diffusion.train_util import TrainLoop
+from utils.models import ModelLoader
+
+
+def main():
+    args = create_argparser().parse_args()
+
+    if (type(args.few_shot_count) not in [int, None]) or (args.few_shot_count is int and args.few_shot_count <= 0):
+        print("few_shot_count argument has to be a non-negative integer if stated.")
```
Comment (talafek96) [inline scripts/train.py:24]:
```suggestion
        print("few_shot_count argument has to be a positive integer if stated.")
```

**Review by talafek96 — APPROVED** [review]

**Pull request merged**